src/tir/schedule/state.cc

```diff
@@ -206,25 +188,11 @@ class StateCreator : private StmtVisitor {
    * \param stmt A for-loop statement or a block statement
    * \return A sref to the stmt
    */
-  StmtSRef PushSRef(const StmtNode* stmt) {
-    if (srefs_.empty()) {
-      srefs_.push_back(
-          StmtSRef(stmt,
-                   /*parent=*/nullptr,
-                   /*seq_index=*/-1));  // `seq_index` will be set properly in SetSeqIndex
-    } else {
-      StmtSRefNode* parent = srefs_.back().get();
-      srefs_.push_back(
-          StmtSRef(stmt, parent,
-                   /*seq_index=*/-1));  // `seq_index` will be set properly in SetSeqIndex
-    }
-    return srefs_.back();
-  }
+  void PushSRef(const StmtNode* stmt) { srefs_.push_back(self_->stmt2ref.at(stmt)); }
 
-  /*! \brief Pop the top of the scope and record it in stmt2ref map */
-  StmtSRef PopAndRecordSRef() {
-    StmtSRef sref = std::move(srefs_.back());
-    self_->stmt2ref[sref->stmt] = sref;
+  /*! \brief Pop the top of the scope */
```

Comment:
If I understand correctly, the `MakeBlockInfo` method may be problematic in terms of the `is_root_block` flag, because `sref_` can be empty on a subtree root instead of the root of the entire tree. Is that correct?

Reply:
Yes. But it doesn't matter.

Reply:
but why? Will we assume the flag for the sub tree root to be true?

Reply:
The flag only affects the info binding calculation, I think?

Reply:
Correct me if I was wrong, but `ScheduleStateNode::UpdateSubtreeblockInfo(stmt)` calls `BlockInfoCollector::Collect`, where `stmt` can be the a root block of a subtree, but it's binding might not be affine; However, in `MakeBlockInfo`, we would assume `is_root_block == True`, and make this binding affine - is that correct?

Reply:
consider `ScheduleStateNode::UpdateSubtreeblockInfo(stmt)`
If stmt is a Block, which means the outer loops of Block are not passed into the function, then we should expect the Block to be root(i.e. it doesn't have outer loops).
If stmt is a Loop, the the blocks under the loop can not be judged as root block.